atlp-rwanda/Phantom-Backend-Pirates
# Ft - Change Password

## PR Introduction
As an authenticated user! You should be able to change password anytime you feel like changing them for any security reason.
## Description of Task to be completed
- Authenticated user should be able to change password
- Appropriate error messages are displayed on unsuccessful attempts
## How should this be manually tested?
- Clone this branch and cd into it
- Install all the packages used by running `npm install`
- Create tables by running in your terminal `npm run migrate`
- Run the server by running in your terminal `npm run dev`
- Using Postman test this change password api using this endpoint `/changepassword`
- To test this effectively try to also use **postman** to login as admin or any other and then change the passwords using the above endpoint by having in your body the following;
```
  email:
  oldPassword:
  newPassword:
  passwordConfirm: 

```
**Once you are done, you can test the login endpoint and see if your password changed**

**Note:** You need to have you .env with the following components in case you don't:

```
HOST_NAME = localhost
AUTH_SECRET = 
AUTH_EXPIRES = 1d
AUTH_ROUNDS = 10 
ADMIN_SECRET = 
OPERATOR_SECRET = 
DRIVER_SECRET = 
RESET_SECRET = 
RESET_TOKEN_EXPIRES = 2hr
EMAIL = phantom@gmail.com
PASSWORD = 
TRANSPORTER_HOST = 0.0.0.0
TRANSPORTER_PORT = 1025
BASE_URL = **(eg. http://locolhost:3000)**
LOGOUT_JWT_EXPIRES = 1s
CHANGE_PASSWORD_SECRET = 
```
